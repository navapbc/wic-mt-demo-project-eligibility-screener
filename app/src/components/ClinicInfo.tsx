export type ClinicInfoProps = {
  name: string
  streetAddress: string
  phone: string
  isFormElement?: boolean
}

export const ClinicInfo = (props: ClinicInfoProps) => {
  const { name, streetAddress, phone, isFormElement = false } = props

  const formClass = isFormElement ? 'usa-checkbox__label-description' : ''

  return (
    <div className="clinic-info">
      <div className="clinic-name">{name}</div>
      <div className={formClass}>
        <div className="clinic-street-address">{streetAddress}</div>
        <div className="clinic-phone">{phone}</div>
      </div>
    </div>
  )
}

export default ClinicInfo
