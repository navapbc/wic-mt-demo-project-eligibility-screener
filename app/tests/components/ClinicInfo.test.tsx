import { render, screen } from '@testing-library/react'

import ClinicInfo from '@components/ClinicInfo'

const testProps = {
  name: 'clinic name',
  streetAddress: 'clinic address',
  phone: 'clinic phone',
  isFormElement: undefined,
}

it('should match snapshot when it is a form element', () => {
  const { container } = render(
    <ClinicInfo {...testProps} isFormElement={true} />
  )
  expect(container).toMatchSnapshot()
})

it('should match snapshot when it is not a form element', () => {
  const { container } = render(
    <ClinicInfo {...testProps} isFormElement={false} />
  )
  expect(container).toMatchSnapshot()
})
