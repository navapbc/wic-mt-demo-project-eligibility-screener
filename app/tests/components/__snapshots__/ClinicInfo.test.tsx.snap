// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when it is a form element 1`] = `
<div
  className="clinic-info"
>
  <div
    className="clinic-name"
  >
    clinic name
  </div>
  <div
    className="usa-checkbox__label-description"
  >
    <div
      className="clinic-street-address"
    >
      clinic address
    </div>
    <div
      className="clinic-phone"
    >
      clinic phone
    </div>
  </div>
</div>
`;

exports[`should match snapshot when it is not a form element 1`] = `
<div
  className="clinic-info"
>
  <div
    className="clinic-name"
  >
    clinic name
  </div>
  <div
    className=""
  >
    <div
      className="clinic-street-address"
    >
      clinic address
    </div>
    <div
      className="clinic-phone"
    >
      clinic phone
    </div>
  </div>
</div>
`;
